modules/services/fnott.nix

```diff
@@ -32,8 +32,8 @@ in {
       };
 
       configFile = mkOption {
```

Comment:
Perhaps could remove this option altogether? If you make the user service depend on the configuration file then activation should restart the service.

Reply:
yes, you are right though I am afraid removing it will not be backwards compatible, I definitely should have given it more thought before including it in the initial module addition.

Reply:
I think in this case removal is OK, the module is not particularly old, most users probably wouldn't have referred to this option explicitly, and if they have then fixing the problem is easy by just removing the option assignment.

I would propose doing something along the lines of

```patch
diff --git a/modules/services/fnott.nix b/modules/services/fnott.nix
index a6a1aa69..e03d0f96 100644
--- a/modules/services/fnott.nix
+++ b/modules/services/fnott.nix
@@ -18,6 +18,12 @@ let
       };
     in attrsOf (attrsOf iniAtom);
 in {
+  imports = [
+    (mkRemovedOptionModule [ "services" "fnott" "configFile" ] ''
+      Support for this option has been removed. Configuration is now read
+      directly from the generated configuration file.'')
+  ];
+
   options = {
     services.fnott = {
       enable = mkEnableOption ''
@@ -31,23 +37,6 @@ in {
         description = "Package providing <command>fnott</command>.";
       };
 
-      configFile = mkOption {
-        type = types.either types.str types.path;
-        default = "${config.xdg.configHome}/fnott/fnott.ini";
-        defaultText = "$XDG_CONFIG_HOME/fnott/fnott.ini";
-        description = ''
-          Path to the configuration file read by fnott.
-          </para><para>
-          Note that environment variables in the path won't be properly expanded.
-          </para><para>
-          The configuration specified under 
-          <option>services.fnott.settings</option> will be generated and
-          written to <filename>$XDG_CONFIG_HOME/fnott/fnott.ini</filename>
-          regardless of this option. This allows using a mutable configuration file
-          generated from the immutable one, useful in scenarios where live reloading is desired.
-        '';
-      };
-
       extraFlags = mkOption {
         type = types.listOf types.str;
         default = [ ];
@@ -101,13 +90,16 @@ in {
         Documentation = "man:fnott(1)";
         After = [ "graphical-session-pre.target" ];
         PartOf = [ "graphical-session.target" ];
+        X-Restart-Triggers = mkIf (cfg.settings != { }) [
+          (toString config.xdg.configFile."fnott/fnott.ini".source)
+        ];
       };
 
       Service = {
         Type = "dbus";
         BusName = "org.freedesktop.Notifications";
         ExecStart = concatStringsSep' " " [
-          "${cfg.package}/bin/fnott -c ${cfg.configFile}"
+          "${cfg.package}/bin/fnott"
           (escapeShellArgs cfg.extraFlags)
         ];
       };
```